h2o-core/src/main/java/water/fvec/TransformWrappedVec.java

```diff
@@ -157,6 +157,27 @@ public Chunk deepCopy() {
     double apply();
   }
 
+  public abstract static class Function1DTransform implements Transform {
+    private double _value;
+    @Override
+    public void reset() {
+      // noop
+    }
+
+    @Override
+    public void setInput(int i, double value) {
+      assert i == 0;
+      _value = value;
+    }
+
+    @Override
+    public double apply() {
+      return apply(_value);
+    }
+
+    public abstract double apply(double x);
```

Comment:
could be just `protected` I guess, given that the "true" functional method is `apply()`

Reply:
thanks, I will make it protected - it only needs to be overridden, it is not "publicly" called